### PR TITLE
Add more logs to debug the output of Sarama `DescribeTopics` method

### PR DESCRIPTION
**Pull request by dttung2905**
_Provide a description of what has been changed_

### Checklist

- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Relates to https://github.com/kedacore/keda/issues/5102

### CODE REVIEW

File: pkg/scalers/kafka_scaler.go [hunk 1]
```diff
@@ -594,6 +594,9 @@ func (s *kafkaScaler) getTopicPartitions() (map[string][]int32, error) {
 	if err != nil {
 		return nil, fmt.Errorf("error describing topics: %w", err)
 	}
+	s.logger.Info(
+		fmt.Sprintf("with topic name %s the list of topic metadata is %v", topicsToDescribe, topicsMetadata),
+	)
```
Comment (JorTurFer) [inline pkg/scalers/kafka_scaler.go:599]: As it's a debug logger, let's change the verbosity:
```suggestion
	s.logger.V(1).Info(
		fmt.Sprintf("with topic name %s the list of topic metadata is %v", topicsToDescribe, topicsMetadata),
	)
```
WDYT @zroubalik ?

Comment (zroubalik) [reply pkg/scalers/kafka_scaler.go:599]: Agree!  Could you please add this to the other kafka scaler that we have? (If possible)

Comment (dttung2905) [reply pkg/scalers/kafka_scaler.go:599]: sure. I was searching for something like `s.logger.Debug()` but could not find any. Thanks for the suggestion

Comment (dttung2905) [reply pkg/scalers/kafka_scaler.go]: @zroubalik  for apache_kafka_scaler, I can see that they have put similar log. Do you want to put the logging anywhere else ? https://github.com/kedacore/keda/blob/4cbc74e51738eae35f9c8f476bdb3bc7fea03982/pkg/scalers/apache_kafka_scaler.go#L423-L429

Comment (zroubalik) [reply pkg/scalers/kafka_scaler.go]: Just make sure we expose the same info when needed and also in the same log level (I can see 4 in this snippet, so make it 1 pls)

Comment (dttung2905) [reply pkg/scalers/kafka_scaler.go]: I have made changes on the apache_kafka_scaler too!

**Comment by dttung2905** [comment]
```
=== RUN   TestPrometheusScalerExecutePromQueryParameters
    prometheus_scaler_test.go:402: Expect request path to = /api/v1/query?query=&time=2023-11-02T22:05:25Z&first=foo&second=bar but it is /api/v1/query?query=&time=2023-11-02T22:05:25Z&second=bar&first=foo
--- FAIL: TestPrometheusScalerExecutePromQueryParameters (0.00s)
```
https://github.com/kedacore/keda/actions/runs/6738632518/job/18318624696?pr=5141
let me check this later. Probably need a rebase

**Comment by dttung2905** [comment]
@zroubalik could you help to run the e2e test again? I rebased to main and all tests passed

**Comment by dttung2905** [comment]
@zroubalik the E2E test still fails. I'm not sure whether its because of the change I introduce or flaky test though. Is there a way to get some logs on this ? :thinking:

**Review by zroubalik — APPROVED** [review]
LGTM

**Pull request merged**